### DISCUSSION TITLE
Phantom App No Longer sending helpful error messages

### DISCUSSION THREAD

**Discussion by jacksturt** [phantom/docs#88] [Help & Support] [answered]
### Summary

In the recent weeks, it seems that phantom error handling has changed from surfacing the errors thrown by a smart contract, so sending a generalized `Unexpected Error`. Is there a new `development` flag or something that needs to be set to surface the full error message? This is painful for debugging, because it forces us to enable `skipPreflight` and set connection config to `{commitment: "finalized"}` to be able to see the errors on chain.

### Steps to Reproduce

1. Interact with any smart contract with a phantom wallet in a way that will cause an error
2. See the `Transaction Simulation Failed` message when trying to sign
3. See that the only error in the console are `Unexpected Error`, no the actual solana/anchor/custom error from the smart contract. Even just surfacing the error # from the smart contract is wildly helpful here

### Phantom Version

23.11.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @jacksturt,

Thanks for writing in with this!

Do you have a hosted example where I can reproduce these errors?

**Comment by jacksturt** [comment]
Hey, sorry for the late reply, we just launched on Tuesday so it's been very busy. The site where this is happening is https://pro.lancer.so/. I would be happy to set up a time to debug this live! We will need to approve your account for you to be able to go through the full reproduction steps. If you want to message me on telegram, @jacksturtevant is my handle. I am much more responsive there!

**Reply by adamdelphantom** [reply]
Hey @jacksturt thanks for the further info.

I've created an account under my email adam dot g phantom dot app. Would you mind approving it, please?

Also does this issue persist with 23.12.0?

**Reply by jacksturt** [reply]
Hey, I approved that email. I will download the new extension and test it out within the next day!

**Reply by jacksturt** [reply]
I just tried with 23.12.0, and the problem persists.

**Reply by adamdelphantom** [reply]
@jacksturt, I'm in. I was able to submit to a quest, w/o issue. Do you have a specific flow I can use to reproduce the unhelpful error message?

**Reply by jacksturt** [reply]
Yes, here is the flow. You will need to accounts to be able to successfully reproduce this. One of which will need to be approved, so please let me know which emails you signed in with:
1. Create a private quest (user 1, wallet 1)
2. Fund that quest (user 1, wallet 1)
3. Send the quest link to user 2
4. Apply to the quest (user 2, wallet 2)
5. Accept the application (user 1, wallet 1)
6. Try to submit to the quest with a different wallet than you applied with (user 2, wallet 3)

This should cause an error since the wallet you applied with is approved, but the new wallet you are using is not approved. Try sending this transaction through, and see that the error message is generic, whereas it should be returning 6009 "You do not have permissions to submit".

I recommend using brave in one tab for user 1, and chrome or a private tab for user 2.

**Comment by jacksturt** [comment]
@adamdelphantom I am still awaiting a response. Could you please schedule a time where I can show you the error live here: https://calendly.com/jack-sturt

**Reply by adamdelphantom** [reply]
Hey @jacksturt sorry for the delay. This is taking time because we need to be able to recreate it outside of your app, so our engineers can investigate. I'm hoping to have it reproducable by the end of week.

**Reply by adamdelphantom** [reply]
@jacksturt Do you have a snippet for the authorization on your smart contract?

**Comment by chinepun** [comment]
@adamdelphantom what my partner @jacksturt is trying to say is previously we could debug why a transaction is failing through the console by getting the Custom Error Code which we can trace to our smart contract but currently we cannot retrieve the Custom Error Code from the console when a transaction is not going through.

**Reply by adamdelphantom** [reply]
Hey @chinepun, Thank you for the further information!

What I'm trying to do is have a reliable way to reproduce the bug you're seeing outside of having to create multiple allowlisted accounts within your application. This is so that anyone on our team can reproduce the bug to investigate and potentially fix. This is the only report I've received of this regression. It would be helpful to have some specific interactions or code for us to be able to recreate locally.

**Comment by jacksturt** [comment]
Hey @adamdelphantom, I added you to our repository that has the smart contract. Let me know if this is sufficient to debug? I have another, not well maintained full stack repo where this issue is happening as well that I can send you, but I don't want to subject you to code I wrote ~a year ago when learning solana. I am getting this error on both, so it may be a FE issue of how I am loading the wallet

**Reply by adamdelphantom** [reply, marked as answer]
@jacksturt There doesn't seem to be anything out of sorts with your anchor errors. Can you provide some context or code on how you are connecting to the wallet?
